dask_planner/src/sql/logical/table_scan.rs

```
@@ -18,6 +19,129 @@ pub struct PyTableScan {
+                    let ident = match *expr.clone() {
+                        Expr::Column(col) => Ok(col.name),
+                        Expr::Literal(val) => Ok(format!("{}", val)),
+                        _ => Err(DaskPlannerError::InvalidIOFilter(format!(
+                            "Invalid InList Expr type `{}`. using in Dask instead",
+                            filter
+                        ))),
+                    };
+
+                    let op = if *negated { "not in" } else { "in" };
+                    let il: Vec<PyObject> = list
+                        .iter()
+                        .map(|f| match f {
+                            Expr::Column(col) => col.name.clone().into_py(py),
+                            Expr::Literal(val) => match val {
+                                ScalarValue::Boolean(val) => val.unwrap().into_py(py),
+                                ScalarValue::Float32(val) => val.unwrap().into_py(py),
+                                ScalarValue::Float64(val) => val.unwrap().into_py(py),
+                                ScalarValue::Int8(val) => val.unwrap().into_py(py),
+                                ScalarValue::Int16(val) => val.unwrap().into_py(py),
+                                ScalarValue::Int32(val) => val.unwrap().into_py(py),
+                                ScalarValue::Int64(val) => val.unwrap().into_py(py),
+                                ScalarValue::UInt8(val) => val.unwrap().into_py(py),
+                                ScalarValue::UInt16(val) => val.unwrap().into_py(py),
+                                ScalarValue::UInt32(val) => val.unwrap().into_py(py),
+                                ScalarValue::UInt64(val) => val.unwrap().into_py(py),
+                                ScalarValue::Utf8(val) => val.clone().unwrap().into_py(py),
+                                ScalarValue::LargeUtf8(val) => val.clone().unwrap().into_py(py),
+                                _ => "".into_py(py),
```

For an unsupported scalar type we should raise an error and have it go via the unfiltered_exprs path